src/test/scala/chiselTests/experimental/DataMirrorSpec.scala

```diff
@@ -55,4 +55,21 @@ class DataMirrorSpec extends ChiselFlatSpec {
     }
     ChiselStage.elaborate(new MyModule)
   }
+
+  it should "support getParent for normal modules" in {
+    class GrandChild(parent: RawModule) extends Module {
+      DataMirror.getParent(this) should be(Some(parent))
+    }
+    class Child(parent: RawModule) extends Module {
+      val inst = Module(new GrandChild(this))
+      DataMirror.getParent(inst) should be(Some(this))
+      DataMirror.getParent(this) should be(Some(parent))
+    }
+    class Parent extends Module {
+      val inst = Module(new Child(this))
+      DataMirror.getParent(inst) should be(Some(this))
+      DataMirror.getParent(this) should be(None)
+    }
+    ChiselStage.elaborate(new Parent)
+  }
```

Comment:
can we show a D/I test. What happens if you call this from within a Definition(...) (sjhould be None I expect) and `getParent(Instance(...))` should be `this`?

Reply:
So this does not support `Definition[_]` nor `Instance[_]`, as I commented below, I'm sort of punting on that possible API extension. I have added a test though.

Reply:
i dont quite get the test, will it just not compile if you try to call it on Definition or Instance?

It does show that it works as expected from within the Definition code though so that's good.